### PR TITLE
Briefly explain how to merge branches

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | n/a |
| Fixed tickets | n/a |

/cc @nicolas-grekas

### CODE REVIEW

File: contributing/code/git.rst [hunk 1]
```diff
@@ -40,3 +40,17 @@ adding ``--show-notes=github-comments`` to the ``git show`` command:
...
+
+Branches
+--------
+
+Branches of lower versions are merged in branches of higher versions on a regular
+basis by the members of the core team.
+
+The ``--log`` option of the ``git merge`` command be used to include the list of
+merged commits in the commit message. For instance, the following command must
+be used to merge the ``3.0`` branch in the ``master`` branch.
```
Comment (xabbuh) [inline contributing/code/git.rst]: I would not include this information in the docs. For the normal user it is not important to know how exactly branches are merged up.

Comment (nicolas-grekas) [inline contributing/code/git.rst]: I think it's still worth documenting, for the openness of the process

**Comment by nicolas-grekas** [comment]
In fact, I added this to my ~/.gitconfig:
```
[merge]
    summary = true
```

I think it's best, because it prevents forgetting about `--log`

File: contributing/code/git.rst [hunk 2]
```diff
@@ -40,3 +40,17 @@ adding ``--show-notes=github-comments`` to the ``git show`` command:
...
 
     $ git show HEAD --show-notes=github-comments
+
+Branches
+--------
+
+Branches of lower versions are merged in branches of higher versions on a regular
+basis by the members of the core team.
+
+The ``--log`` option of the ``git merge`` command be used to include the list of
```
Comment (javiereguiluz) [inline contributing/code/git.rst]: A word is probably missing after `command`:

`command be used` -> `command can be used` ? `command must be used` ?

Comment (nicolas-grekas) [inline contributing/code/git.rst]: ...`command is used`...

File: contributing/code/git.rst (hunk 1 above)
Comment (xabbuh) [inline contributing/code/git.rst]: I am not against transparency, but we need to be careful to not confuse new contributors. We already provide a lot of information and adding some more will not make it easier for new contributors to find the information that is really necessary for them.

**Comment by WouterJ** [comment]
I agree with @xabbuh here, I think it's not usefull information for non-mergers. If mergers forget to pass `--log`, an internal file should be the solution.

As multiple doc reviewers are against this change, I'm closing this PR. If you don't agree, feel free to comment :)

**Comment by stof** [comment]
If we include this, it should at least be in a different location (a page describing the maintenance process is a better fit than the contributing page)

**Pull request closed without merging**